common/rpc/interceptor/namespace_validator.go

```diff
@@ -92,6 +96,10 @@ func (ni *NamespaceValidatorInterceptor) LengthValidationIntercept(
 	info *grpc.UnaryServerInfo,
 	handler grpc.UnaryHandler,
 ) (interface{}, error) {
+	err := ni.setNamespaceIfNotPresent(ni.namespaceRegistry, req)
+	if err != nil {
+		return nil, err
```

Comment:
@yiminc @yycptt what if customer modify & put a namespace ID in the task token?

if returning not found error here legit?

Reply:
Existing behavior will also return not found in this case I think?